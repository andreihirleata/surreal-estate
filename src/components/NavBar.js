import React from "react";
import { Link } from "react-router-dom";
import { FontAwesomeIcon } from "@fortawesome/react-fontawesome";
import { faFortAwesomeAlt } from "@fortawesome/free-brands-svg-icons";

import "../styles/NavBar.css";

const NavBar = () => {
  return (
    <div className="nav-bar">
      <FontAwesomeIcon
        icon={faFortAwesomeAlt}
        style={{ fontSize: "35px", color: "white" }}
      />
      <div className="link-wrapper">
        <ul className="navbar-links">
          <Link to="/">
            <li className="navbar-links-item">View Properties</li>
          </Link>

          <Link to="/add-property">
            <li className="navbar-links-item">Add Property</li>
          </Link>
        </ul>
      </div>
    </div>
  );
};

export default NavBar;
