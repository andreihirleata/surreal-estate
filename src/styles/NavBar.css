.nav-bar {
  background-color: white;
  border-bottom: 4px solid grey;
}

.surreal-logo {
  height: 70px;
  width: 200px;
}

.navbar-links {
  margin: auto;
  padding-top: 10px;
  padding-bottom: 10px;
  font-size: 20px;
  background-color: purple;
  height: 30px;
  display: flex;
  justify-content: flex-start;
}

.navbar-links-item {
  margin: auto;
  padding: 5px 15px;
  display: inline-block;
  color: #fff;
  height: 30px;
}
