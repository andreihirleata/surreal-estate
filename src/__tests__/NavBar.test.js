import React from "react";
import { render } from "@testing-library/react";
import NavBar from "../components/NavBar";
import { createMemoryHistory } from "history";
import { Router } from "react-router-dom";

test("renders properties component", () => {
  const history = createMemoryHistory();
  const { asFragment } = render(
    <Router history={history}>
      <NavBar />
    </Router>
  );
  const component = asFragment();
  expect(component).toMatchSnapshot();
});
