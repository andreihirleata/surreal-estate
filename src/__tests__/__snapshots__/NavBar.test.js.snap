// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`renders properties component 1`] = `
<DocumentFragment>
  <div
    class="nav-bar"
  >
    <img
      alt="logo"
      class="surreal-logo"
      src="https://mcrcodes.s3.eu-west-2.amazonaws.com/course/surreal-estate/logo.png"
    />
    <ul
      class="navbar-links"
    >
      <a
        href="/"
      >
        <li
          class="navbar-links-item"
        >
          View Properties
        </li>
      </a>
      <a
        href="/add-property"
      >
        <li
          class="navbar-links-item"
        >
          Add a Property
        </li>
      </a>
    </ul>
  </div>
</DocumentFragment>
`;
